packages/playwright-test/src/store.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright Microsoft Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs';
+import path from 'path';
+import type { TestStore } from '../types/test';
+import { currentTestInfo } from './common/globals';
+
+class JsonStore implements TestStore {
+  private _toFilePath(name: string) {
+    const testInfo = currentTestInfo();
+    if (!testInfo)
+      throw new Error('store can only be called while test is running');
```

Comment:
We should also allow it during global setup. Perhaps even during describe, for test generation? Maybe no restrictions?

Reply:
Let's address it separately.